src/test/java/com/virtuslab/archunit/ClassStructureTestSuite.java

```diff
@@ -142,4 +166,23 @@ public boolean test(JavaConstructorCall input) {
             "and passing a reference to the enclosing object (or to the fields thereof) explicitly")
         .check(importedClasses);
   }
+  @Test
+  public void no_classes_declaring_LOG_field_should_call_log_method() {
+    noClasses()
+        .that(new DescribedPredicate<JavaClass>("declare `LOG` field") {
+          @Override
+          public boolean test(JavaClass javaClass) {
+            return javaClass.getFields().stream().anyMatch(field -> field.getName().equals("LOG"));
+          }
+        })
+        .should()
+        .callMethodWhere(new DescribedPredicate<JavaMethodCall>("call `log` method") {
+          @Override
+          public boolean test(JavaMethodCall javaMethodCall) {
+            return javaMethodCall.getTarget().getName().equals("log");
+          }
+        })
+        .because("LOG field should be used explicitly instead")
```

Comment:
oh, I guess that's the answer to one of my former questions... but y?

Reply:
...and here just a matter of style (using `log()` isn't a mistake, just an unnecessary indirection)

Reply:
okok, makes sense ! 🪨et